metadata-ingestion/tests/unit/test_bigquery_source.py

```diff
@@ -173,3 +169,158 @@ def test_upstream_table_generation_with_temporary_table_with_multiple_temp_upstr
+    config = BigQueryConfig.parse_obj(
+        {
+            "profiling": {
+                "profile_if_updated_since_days": 1,
+                "profile_table_size_limit": 5,
+                "profile_table_row_limit": 50000,
+            }
+        }
+    )
+    source = BigQuerySource(config=config, ctx=PipelineContext(run_id="test"))
+    threshold_time = datetime.fromisoformat("2022-04-02 10:42:29")
+    expected_query = (
+        "SELECT table_id, size_bytes, last_modified_time, row_count, FROM dataset_foo.__TABLES__ WHERE "
+        "row_count<50000 and ROUND(size_bytes/POW(10,9),2)<5 and last_modified_time>=1648876349000 "
+    )
+    query = source.generate_profile_candidate_query(threshold_time, "dataset_foo")
+    assert query == expected_query
+
+
+def test_bq_get_profile_candidate_query_no_day_limit():
+    config = BigQueryConfig.parse_obj(
+        {
+            "profiling": {
+                "profile_if_updated_since_days": None,
+                "profile_table_size_limit": 5,
+                "profile_table_row_limit": 50000,
+            }
+        }
+    )
+    source = BigQuerySource(config=config, ctx=PipelineContext(run_id="test"))
+    expected_query = (
+        "SELECT table_id, size_bytes, last_modified_time, row_count, FROM dataset_foo.__TABLES__ WHERE "
+        "row_count<50000 and ROUND(size_bytes/POW(10,9),2)<5 "
+    )
+    query = source.generate_profile_candidate_query(None, "dataset_foo")
+    assert query == expected_query
+
+
+def test_bq_get_profile_candidate_query_no_size_limit():
+    config = BigQueryConfig.parse_obj(
+        {
+            "profiling": {
+                "profile_if_updated_since_days": 1,
+                "profile_table_size_limit": None,
+                "profile_table_row_limit": 50000,
+            }
+        }
+    )
+    source = BigQuerySource(config=config, ctx=PipelineContext(run_id="test"))
+    threshold_time = datetime.fromisoformat("2022-04-02 10:42:29")
+    expected_query = (
+        "SELECT table_id, size_bytes, last_modified_time, row_count, FROM dataset_foo.__TABLES__ WHERE "
+        "row_count<50000 and last_modified_time>=1648876349000 "
+    )
+    query = source.generate_profile_candidate_query(threshold_time, "dataset_foo")
+    assert query == expected_query
+
+
+def test_bq_get_profile_candidate_query_no_row_limit():
+    config = BigQueryConfig.parse_obj(
+        {
+            "profiling": {
+                "profile_if_updated_since_days": 1,
+                "profile_table_size_limit": 5,
+                "profile_table_row_limit": None,
+            }
+        }
+    )
+    source = BigQuerySource(config=config, ctx=PipelineContext(run_id="test"))
+    threshold_time = datetime.fromisoformat("2022-04-02 10:42:29")
+    expected_query = (
+        "SELECT table_id, size_bytes, last_modified_time, row_count, FROM dataset_foo.__TABLES__ WHERE "
+        "ROUND(size_bytes/POW(10,9),2)<5 and last_modified_time>=1648876349000 "
```

Comment:
So we are saying 1GB = 10^9 bytes. We could also use 1GB = 1024^3 bytes = 2 ^30 bytes. approximately same but some difference. Ok if the choice is explicitly made.